src/main/scala/io/snyk/plugin/depsource/DepTreeProvider.scala

```diff
@@ -1,41 +1,114 @@
 import io.snyk.plugin.datamodel.SnykMavenArtifact
-import org.jetbrains.idea.maven.project.{MavenProject, MavenProjectsManager}
+import org.jetbrains.idea.maven.project.MavenProjectsManager
+import org.jetbrains.plugins.gradle.settings.GradleSettings
+import org.jetbrains.plugins.gradle.util.GradleConstants
+
 import scala.collection.JavaConverters._
 
 trait DepTreeProvider {
   def rootIds: Seq[String]
   def getDepTree(rootId: String): Option[SnykMavenArtifact]
-  def idToMavenProject(id: String): Option[MavenProject]
+  def idToBuildToolProject(id: String): Option[BuildToolProject]
 }
 
 private class ProjectDepTreeProvider(project: Project) extends DepTreeProvider {
-  private[this] def allMavenProjects: Seq[MavenProject] =
-    MavenProjectsManager.getInstance(project).getProjects.asScala
 
-  override def rootIds: Seq[String] = allMavenProjects.map(_.toString)
+  private[this] def isMavenProject: Boolean = !MavenProjectsManager.getInstance(project).getProjects.isEmpty
+
+  private[this] def isGradleProject: Boolean = !GradleSettings.getInstance(project).getLinkedProjectsSettings.isEmpty
+
+  private[this] def getMavenBuildToolProjects: Seq[BuildToolProject] = {
+    val mavenProjects = MavenProjectsManager.getInstance(project).getProjects.asScala
+
+    mavenProjects.map(mavenProject => {
+      MavenBuildToolProject(mavenProject)
+    })
+  }
+
+  private[this] def getGradleBuildToolProjects: Seq[BuildToolProject] = {
+    val gradleProjects = GradleSettings.getInstance(project).getLinkedProjectsSettings.asScala
+
+    gradleProjects.map(gradleSettings => {
+      val projectPath = gradleSettings.getExternalProjectPath
+
+      val projectData = findProjectData(project, GradleConstants.SYSTEM_ID, projectPath)
+
+      val gradleModuleData = ExternalSystemApiUtil.find(projectData, ProjectKeys.MODULE)
+
+      GradleBuildToolProject(gradleModuleData.getData)
+    }).toList
+  }
+
+  private[this] def getAllBuildToolProjects: Seq[BuildToolProject] = {
+    if (isMavenProject) {
+      getMavenBuildToolProjects
+    } else if (isGradleProject) {
+      getGradleBuildToolProjects
+    } else {
+      println("Unsupported (for now) project type.")
+
+      Seq()
+    }
+  }
+
+  override def rootIds: Seq[String] = getAllBuildToolProjects.map(_.toString)
 
-  override def idToMavenProject(id: String): Option[MavenProject] =
-    allMavenProjects.find(_.toString == id)
+  override def idToBuildToolProject(id: String): Option[BuildToolProject] =
+    getAllBuildToolProjects.find(_.toString == id)
 
   override def getDepTree(rootId: String): Option[SnykMavenArtifact] = {
-    val projects = allMavenProjects
-    val mp = idToMavenProject(rootId) orElse projects.headOption
-    mp map SnykMavenArtifact.fromMavenProject
+    val projects = getAllBuildToolProjects
+    val maybeBuildToolProject = idToBuildToolProject(rootId) orElse projects.headOption
+
+    maybeBuildToolProject map SnykMavenArtifact.fromBuildToolProject
```

Comment:
This code is from the very first version. I didn't particularly change it. Easier to say it get SnykMavenArtifact for the project. But it seems it can be changed.